web/packages/teleport/src/Console/DocumentSsh/DocumentSsh.tsx

```diff
@@ -58,21 +61,28 @@ export default function DocumentSsh({ doc, visible }: PropTypes) {
 
   return (
     <Document visible={visible} flexDirection="column">
-      <FileTransferContextProvider>
-        <FileTransferActionBar isConnected={doc.status === 'connected'} />
-        {status === 'loading' && (
-          <Box textAlign="center" m={10}>
-            <Indicator />
-          </Box>
-        )}
-        {webauthn.requested && (
-          <AuthnDialog
-            onContinue={webauthn.authenticate}
-            onCancel={closeDocument}
-            errorText={webauthn.errorText}
+      <FileTransferActionBar isConnected={doc.status === 'connected'} />
+      {status === 'loading' && (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      )}
+      {webauthn.requested && (
+        <AuthnDialog
+          onContinue={webauthn.authenticate}
+          onCancel={closeDocument}
+          errorText={webauthn.errorText}
+        />
+      )}
+      {status === 'initialized' && <Terminal tty={tty} ref={refTerminal} />}
+      <FileTransferContainer>
```

Comment:
This `FileTransferContainer` is necessary because we want both `FileTransferRequests` and `FileTransferStateless` to be render within a `position: absolute` div, right? Would there ever be a callsite where `FileTransferRequests` is used without `FileTransfer`?

In my mind, the two are closely coupled together. With this API, I feel like we're running into the same problem as the one I initially pointed out with `useFilesStore`, that is "Hey, if you want to use `FileTransfer` or `FileTransferRequests`, you actually have to wrap them in `FileTransferContainer` and there is nothing to warn you if you forget to do so".

This might be a bit unorthodox ([idk, maybe?](https://epicreact.dev/one-react-mistake-thats-slowing-you-down/)), but if we assume that…

* We always want to render the file transfer requests together with the file transfer form.
* `FileTransfer` needs to be reusable between the Web UI and Connect, but only the Web UI has to support requests at the moment.

…then what if `FileTransfer` accepted `FileTransferRequests` as a prop (`Requests` or `$requests`, I don't know if we have an established convention for this) and then it'd render that prop next to `FileTransferDialog` if present?

Reply:
This was also decided originally when trying to use `filesStore.start` before the context was moved. I think passing it as a prop to FileTransfer would work perfectly. We have somewhat of a precedent (of passing component as prop) in our `SearchPanel` but now I can't seem to find it. I prefer `Requests` over `$requests` but have no argument besides aesthetics :)

I'll refactor this out and think it'll be much nicer!